packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -134,89 +132,99 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
+            let modelName = try FlutterDataStoreRequestUtils.getModelName(args: args)
+            let modelSchema = try FlutterDataStoreRequestUtils.getModelSchema(modelSchemas: flutterModelRegistration.modelSchemas, modelName: modelName)
+            let serializedModelData = try FlutterDataStoreRequestUtils.getSerializedModelData(args: args)
+            let modelID = try FlutterDataStoreRequestUtils.getModelID(serializedModelData: serializedModelData)
             
-            guard let modelSchema = flutterModelRegistration.modelSchemas[modelName] else {
-                throw DataStoreError.decodingError("Unable to get model from registered schemas", "Check the model name.")
+            var queryPredicate: QueryPredicate?
+            if let queryPredicateData = args["queryPredicate"] as? [String: Any] {
+                queryPredicate = try QueryPredicateBuilder.fromSerializedMap(queryPredicateData)
             }
             
-            try bridge.onDelete(id: id,
-                              modelData: modelData,
-                              modelSchema: modelSchema) { (result) in
+            let serializedModel = SerializedModel(id: modelID, map: try FlutterDataStoreRequestUtils.getJSONValue(serializedModelData))
+           
+            try bridge.onSave(
+                serializedModel: serializedModel,
+                modelSchema: modelSchema,
+                when: queryPredicate
+            ) { (result) in
                 switch result {
                 case .failure(let error):
-                    print("Delete API failed. Error = \(error)")
-                    FlutterDataStoreErrorHandler.handleDataStoreError(error: error,
-                                                                      flutterResult: flutterResult,
-                                                                      msg: FlutterDataStoreErrorMessage.DELETE_FAILED.rawValue)
-                case .success():
+                    print("Save API failed. Error: \(error)")
+                    FlutterDataStoreErrorHandler.handleDataStoreError(
+                        error: error,
+                        flutterResult: flutterResult,
+                        msg: FlutterDataStoreErrorMessage.SAVE_FAILED.rawValue
+                    )
+                case .success(let model):
+                    print("Successfully saved model: \(model)")
                     flutterResult(nil)
                 }
             }
-            
-        } catch {
-            print("Failed to parse delete arguments with \(error)")
+        }
+        catch let error as DataStoreError {
+            print("Failed to parse save arguments with \(error)")
+            FlutterDataStoreErrorHandler.handleDataStoreError(
+                error: error,
+                flutterResult: flutterResult,
+                msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue
+            )
+        }
```

Comment:
Do we need any specific error handling here? Didn't we decide to not throw an error if query predicate causes the update to not go through?